### PR TITLE
re-enable the PDB eviction test

### PR DESCRIPTION
**Pull request by tzneal**
**1. Issue, if available:**
#638

**2. Description of changes:**

- the pod must be in a non-pending state for eviction  to consider the PDB
- the Status.ObservedGeneration must be >= the pdb.Generation  or the eviction results in a 429 retry with a 10 second delay

**3. How was this change tested?**

Unit testing

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [X] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/termination/suite_test.go [hunk 1]
```diff
@@ -204,9 +204,17 @@ var _ = Describe("Termination", func() {
...
 			})
 
-			ExpectCreated(ctx, env.Client, node, podNoEvict, pdb)
+			ExpectCreated(ctx, env.Client, node, pdb)
+			ExpectCreatedWithStatus(ctx, env.Client, podNoEvict)
+
+			// kube-controller-manager normally does this, but we don't have one. If this isn't modified
+			// the eviction controller assumes that the PDB hasn't been processed by the disruption controller
+			// yet and adds a 10 second retry to our evict() call
+			pdb.Status.ObservedGeneration = 1
```
Comment (ellistarn) [inline pkg/controllers/termination/suite_test.go:216]: Can we include this in the `test.PodDisruptionBudget` library? We will then need to use ExpectCreatedWithStatus on it.

Comment (tzneal) [reply pkg/controllers/termination/suite_test.go]: Yup, done, the update call required the `ResourceVersion` to match, I thought that was with everything so I'm not sure why it only showed up for the PDB.

File: pkg/test/pods.go [hunk 2]
```diff
@@ -126,6 +128,13 @@ func PodDisruptionBudget(overrides ...PDBOptions) *v1beta1.PodDisruptionBudget {
 			},
 			MaxUnavailable: options.MaxUnavailable,
 		},
+		Status: v1beta1.PodDisruptionBudgetStatus{
+			// To be considererd for application by eviction, the Status.ObservedGeneration must be >= the PDB generation.
```
Comment (ellistarn) [inline pkg/test/pods.go:132]:
```suggestion
			// To be considered for application by eviction, the Status.ObservedGeneration must be >= the PDB generation.
```

**Review by ellistarn — DISMISSED** [review]

File: pkg/test/pods.go (hunk 2 above)
Comment (tzneal) [reply pkg/test/pods.go]: Good catch, done.

**Review by ellistarn — APPROVED** [review]

**Pull request merged**